### ISSUE TITLE
pkgin upgrade crashes with second repository

### ISSUE THREAD

**Issue by Licenser** [TritonDataCenter/pkgsrc-legacy#32]
Hi I'm using a second repository in `pkgin`.
And it seems that with this repository `pkgin` crashes during the call of `pkgin upgrade`.

/opt/local/etc/pkgin/repositories.conf:

```
http://pkgsrc.joyent.com/sdc6/2012Q2/x86_64/All
http://release.project-fifo.net/pkg/dev/
```

I tested the following:
1) Having the default repo twice doesn't causes a crash.
2) The order of repos doesn't matter.
3) This isn't triggered when no packages are installed.

It might be a fault in the package description I'll look into that detail a bit later (packages in the 2nd repo are mine). So they do install fine, and I guess even with a missing field they should not SegFault the program :)

The coredump is here is the following: https://www.dropbox.com/s/l34sh3agxf4r1n6/core

**Comment by cconrad** [comment]
Experiencing the same issue with "pkgin 0.6.3.1 for SunOS-5.11 x86_64 (using SQLite 3.7.15)". Has this been fixed?

**Comment by cconrad** [comment]
Workaround:
1. Remove the second repository from /opt/local/etc/pkgin/repositories.conf
2. Run pkgin -f up

**Comment by mamash** [comment]
Simple: there's no official support for multiple repositories in pkgin (yet).

**Comment by jacques** [comment]
Filip: any idea when pkgin with officially support multiple repositories?

**Comment by mamash** [comment]
No idea. imil had it on his "list of things to do this summer" this year, but I haven't seen any signs of that list being worked on at all.

**Comment by iMilnb** [comment]
I'm trying to reproduce that behavior, unfortunately your repository is for x86_64 and I only have my hands on an i386 SmartOS zone, any chance you could build a repository for i386?

**Comment by cconrad** [comment]
Hi iMilnb,

sorry, I am not using SmartOS anymore, thus I am not able to help with this :/

Best regards,
Claus Conrad

**Comment by bahamat** [comment]
Root cause is NetBSDfr/pkgin#1 which has now been fixed. As soon as we get the new version of pkgin imported this issue can be closed.

**Comment by rmustacc** [comment]
Appears to be an erroneous github auto-close.

**Comment by bahamat** [comment]
This was fixed in commit NetBSDfr/pkgin@338d0dd (see NetBSDfr/pkgin#1).
